deploy/kubernetes/overlays/prow-gke-release-staging-rc/kustomization.yaml

```diff
@@ -21,3 +21,14 @@ images:
 - name: gke.gcr.io/csi-snapshotter
   newName: gcr.io/gke-release-staging/csi-snapshotter
   newTag: "v2.1.1-gke.0"
+
+resources:
+- leader-election-rbac.yaml
+
+patchesJson6902:	
+- target:	
+    group: apps	
+    version: v1	
+    kind: StatefulSet	
```

Comment:
Do we want to switch to a Deployment as well? I think it's not easily done without modifying the base too

Reply:
I'm leaning towards optimizing for the common case of 1 replica. I think this is the common case, but if we have indicators otherwise we could change it

Reply:
Maybe let's set the pod management policy to "parallel" then? That way the statefulset won't get stuck if one of the replicas fail.

Reply:
Actually looking at the [design doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md#cluster-level-deployment) again, StatefulSet is good for a dedicated node, which is probably usually not the case for this deployment. I can switch it to Deployment.

At some point I managed to change the `kind` with a Kustomize JSON patch. Will see if I can still do it :)

Reply:
I think this low risk enough to just do it in the base. There's probably other spec differences between StatefulSet and Deployment too.

Reply:
Got the patching figured out. I pushed a new version

I believe the default of maxUnavailable = 25% (rounded down, so 0) and maxSurge = 25% (rounded up, so 1) is OK - a new replica will be created first, in standby, before the old replica is deleted.